websites/D/Deconreconstruction/presence.ts

```diff
@@ -0,0 +1,153 @@
+					break;
+
+				case "log":
+					presenceData.details = "Viewing the Vast Error log";
+					break;
+
+				default:
+					presenceData.details = "Reading Vast Error";
+					presenceData.smallImageKey = "ve";
+					presenceData.state = `Page ${pathArr[2]} of ${vePages}`;
+					presenceData.smallImageText =
+						document.querySelector("#command").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.deconreconstruction.com${pathname}`,
+						},
+					];
+					break;
+			}
+			break;
+
+		case "thaumatrope":
+			switch (pathArr[2]) {
+				case "syzygy":
+					presenceData.details = "Reading Thaumatrope: Syzygy";
+					presenceData.smallImageKey = "thaumatrope";
+					presenceData.state = `Page ${pathArr[3]} of 162`;
+					presenceData.smallImageText =
+						document.querySelector("#command").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.deconreconstruction.com${pathname}`,
+						},
+					];
+					break;
+
+				case "haustoria":
+					presenceData.details = "Reading Thaumatrope: Haustoria";
+					presenceData.smallImageKey = "thaumatrope";
+					presenceData.state = `Page ${pathArr[3]} of ${htPages}`;
+					presenceData.smallImageText =
+						document.querySelector("#command").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.deconreconstruction.com${pathname}`,
+						},
+					];
+					break;
+
+				case "log":
+					presenceData.details = "Viewing the Thaumatrope log";
+					break;
+
+				default:
+					presenceData.details = "Viewing an unsupported page";
+					break;
+			}
+			break;
+
+		case "bonus":
+			switch (pathArr[2]) {
+				case "april-fools-day-2017":
+					presenceData.details = "Reading April Fools 2017";
+					presenceData.smallImageKey = "mspa";
+					presenceData.state = `Page ${pathArr[3]} of 4`;
+					presenceData.smallImageText =
+						document.querySelector("#command").textContent;
+					presenceData.buttons = [
+						{
+							label: "Read Along",
+							url: `https://www.deconreconstruction.com${pathname}`,
+						},
+					];
+					break;
+
+				case "recap":
+					presenceData.details = "Viewing the Act 1 recap";
+					break;
+
+				case "revised":
+					presenceData.details = "Viewing revised pages";
+					break;
+
+				default:
+					presenceData.details = "Viewing bonus content";
+					break;
+			}
+			break;
+
+		default:
+			presenceData.details = "Viewing an unsupported page";
+			break;
+	}
+
+	if (presenceData.details) presence.setActivity(presenceData);
+	else presence.setActivity();
+});
+
+let vePages: string, htPages: string;
+(function () {
+	function getPages() {
+		if (document.location.pathname.match(/^\/vasterror\/\d+$/)) {
+			fetch(
+				"https://api.deconreconstruction.com/pages/count?story.name=vast-error&published_at_null=false"
+			)
+				.then(res => res.json())
+				.then(res => {
+					vePages = res;
+				});
```

Comment:
Both of the `.then()`s are not needed as you can merge them together

Reply:
how would you recommend to merge them? i keep getting type errors 
![image](https://user-images.githubusercontent.com/38714062/184245563-a3ffc6cb-cfc9-427a-b8d6-51c011500e20.png)
![image](https://user-images.githubusercontent.com/38714062/184245644-c3ec280d-95b0-46ec-b63b-eb33e2e622ed.png)

Reply:
await?

Reply:
Yeah, await

Reply:
i figured as much, but i dont know where to put the await without it telling me this
![image](https://user-images.githubusercontent.com/38714062/184448210-c4cdd89b-a819-4f6c-b679-09931e43082f.png)
(even when i make the function async)

Reply:
Transform your function into an async one.

Reply:
it still marks the await as out of place when making either/both of the functions that its within as async

Reply:
Works for me, try to start an auto-fix, it should be proposed